File: ClientINes/src/app/core/contracts/feedback.ts
import { FeedbackType } from '../enums/feedback-type.enum';

export interface FeedbackItem {
  id: string;
  userName: string;
  userEmail: string;
  type: FeedbackType;
  message: string;
  rating?: number;
  missingFeatures?: string;
  createdAt: string;
  isProcessed: boolean;
}

export interface PagedFeedbackResult {
  items: FeedbackItem[];
  totalCount: number;
  page: number;
  pageSize: number;
}

File: ServerINes/Data/Entities/Infrastructure/Feedback.cs
using INest.Data.Enums;

namespace INest.Data.Entities.Infrastructure
{
    public class Feedback
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid UserId { get; set; }
        public virtual AppUser User { get; set; } = null!;

        public FeedbackType Type { get; set; }
        public string Message { get; set; } = string.Empty;

        public int? Rating { get; set; }
        public string? MissingFeatures { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public bool IsProcessed { get; set; } = false;
    }
}
